module/instances.go

```diff
@@ -0,0 +1,69 @@
+package module
+
+import (
+	"fmt"
+
+	"github.com/emersion/maddy/config"
+	"github.com/emersion/maddy/log"
+)
+
+var (
+	Instances = make(map[string]struct {
+		mod Module
+		cfg *config.Map
+	})
+
+	Initialized = make(map[string]bool)
```

Comment:
I'd prefer this to be unexported and expose a getter instead (`IsInitialized`?)

Reply:
And then a setter too because maddy.Start needs to set status for modules it initializes.

Reply:
Err. Okay, then it's fine to export this one, I guess